src/__tests__/test-suite.ts

```diff
@@ -4,230 +4,227 @@ import { suite } from "../suite"
-       */
-      @testDecorator({
-        name: (param) => `some name #${param}`,
-        params: [
-          { a: 1, b: 2 },
-        ],
-      })
-      protected testSomethingInTheExtendingClass(...args: any[]) { return }
-    }
-    expect(mockDescribe).toHaveBeenCalledTimes(2)
-    mockDescribe.mock.calls[0][1]()
-    mockDescribe.mock.calls[1][1]()
-    // There should be two calls because the parent should not execute the childs test
-    // and the child should not execute the parent tests
-    expect(mockIt).toHaveBeenCalledTimes(2)
-    expect(mockIt.mock.calls).toMatchSnapshot()
-  })
-  test("the child to inherit the beforeAlls from the parent without @suite", () => {
-    const mockTestImpl = jest.fn()
-    class SuperA {
-      public initialConditions: string
-       /**
-        * This beforeAll should be considered even though there is no suite
-        */
-      @beforeAllDecorator
-      public setInitialConditionsOnParent() {
-        this.initialConditions = "initial value"
-      }
-    }
-    @suite
-    class A extends SuperA {
-      /**
-       * This test should only be executed in the child
-       */
-      @beforeAllDecorator
-      public setInitialConditionsOnInstance() {
-        this.initialConditions += " child"
-      }
-      @testDecorator
-      public runTestWithPresetConditions() {
-        mockTestImpl(this.initialConditions)
-      }
-    }
-    expect(mockDescribe).toHaveBeenCalledTimes(1)
-    mockDescribe.mock.calls[0][1]()
-    expect(mockBeforeAll).toHaveBeenCalledTimes(2)
-    mockBeforeAll.mock.calls[0][0]()
-    mockBeforeAll.mock.calls[1][0]()
-    expect(mockIt).toHaveBeenCalledTimes(1)
-    mockIt.mock.calls[0][1](new A())
-    expect(mockTestImpl).toHaveBeenCalledTimes(1)
-    expect(mockTestImpl.mock.calls[0][0]).toBe("initial value child")
-  })
+            @beforeAllDecorator
+            public setInitialConditionsOnInstance() {
+                this.initialConditions = "initial value"
+            }
+            @testDecorator
+            public runParentTestWithPresetConditions() {
+                mockTestImplSuper(this.initialConditions)
+            }
+        }
+
+        @suite
+        class A extends SuperA {
+            @beforeAllDecorator
+            public setInitialConditionsOnInstance() {
+                this.initialConditions += " inherited"
+            }
+            @testDecorator
+            public runTestWithPresetConditions() {
+                mockTestImpl(this.initialConditions)
+            }
+        }
+        expect(mockDescribe).toHaveBeenCalledTimes(2)
+        mockDescribe.mock.calls[0][1]()
+        mockDescribe.mock.calls[1][1]()
+        expect(mockBeforeAll).toHaveBeenCalledTimes(3)
+        mockBeforeAll.mock.calls[0][0]()
+        mockBeforeAll.mock.calls[1][0]()
+        mockBeforeAll.mock.calls[2][0]()
+        // The beforeAll in SuperA is called twice in SuperA
+        // and A and the beforeAll in A is called only in A .
```

Comment:
There is a space before the `.` here and it'd be cool if you'd wrap code-words in markdown style backticks:

```
// The `beforeAll` in `SuperA` is called twice in `SuperA`
// and `A` and the `beforeAll` in `A` is called only in `A`.
// Thus there should be three `beforeAll` calls.
```